src/lib/warmane_client/client.ts

```diff
@@ -63,3 +63,25 @@ export async function getCharacterProfile(params: {
   );
   return response.data;
 }
+
+export async function checkCharacterExists(params: {
+  name: string;
+  realm: string;
+}): Promise<boolean> {
+  const { name, realm } = params;
+  const userAgent = randomUserAgent();
+  const response = await axios.get<string>(
+    `http://armory.warmane.com/character/${name}/${realm}/summary`,
+    {
+      headers: {
+        "User-Agent": userAgent,
+      },
+    }
+  );
+  if (
+    response.data.includes("The character you are looking for does not exist")
```

Comment:
It's probably more likely that the server gets shut down than for them to change or remove "The character you are looking for does not exist", but I think we could make the app a bit more resilient if we found and used another indicator than a specific string.
Teddy Roosevelt once said “Complaining about a problem without posing a solution is called whining.”
I guess Teddy Roosevelt would say that I'm whining, but I'm like 90% sure he's dead.

Reply:
The other issue with this code is that there is no test written for it. I'm not even completely sure it works like I think it does.

I'd like to rely on something that looks more like a 404 response code, but that url always returns 200.

The character profile API would be a better choice, but there's too much rate limiting on it for it to be viable.

If you find a better solution, please let me know.